### PR TITLE
[graph_trainer] Add MXFP8 and Float8 debugmodel configs

### PR DESCRIPTION
**Pull request by aditvenk**
Stacked PRs:
 * __->__#3198
 * #3197

--- --- ---

### [graph_trainer] Add MXFP8 and Float8 debugmodel configs

Add graph_trainer debug-model configs for MXFP8 on Llama 3 and DeepSeek v3, plus Float8 graph_trainer debug-model configs for Llama 3 and DeepSeek v3 EP.

Keep the DeepSeek v3 Float8 EP config local to graph_trainer: it starts from deepseek_v3_debugmodel_ep(), swaps in Float8 linear and grouped-expert converters, then enables the graph_trainer compile config. Also add 4-GPU H100 graph_trainer Float8 integration cases for AOT FX trace with and without full Inductor, and install torchao nightly in the graph_trainer H100 workflow so those cases can run.

Co-Authored with Claude

### CODE REVIEW

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py [hunk 1]
```diff
@@ -35,6 +40,31 @@ def graph_trainer_deepseek_v3_debugmodel_ep() -> GraphTrainer.Config:
...
+def graph_trainer_deepseek_v3_debugmodel_mxfp8() -> GraphTrainer.Config:
+    base = deepseek_v3_debugmodel()
+    base.model_spec = dsv3_model_registry(
+        "debugmodel",
+        moe_comm_backend="standard",
+        quantization=[
+            MXFP8LinearConverter.Config(
+                model_compile_enabled=True,
+                # Include-list of FQN substrings. Skips wkv_a/wkv_b (K/V
+                # projections), router.gate, and lm_head.
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:65]: any particular reasons, these are skipped? 
document heuristic if any.

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py [hunk 2]
```diff
@@ -35,6 +40,31 @@ def graph_trainer_deepseek_v3_debugmodel_ep() -> GraphTrainer.Config:
...
 
 
+def graph_trainer_deepseek_v3_debugmodel_mxfp8() -> GraphTrainer.Config:
+    base = deepseek_v3_debugmodel()
+    base.model_spec = dsv3_model_registry(
+        "debugmodel",
+        moe_comm_backend="standard",
+        quantization=[
+            MXFP8LinearConverter.Config(
+                model_compile_enabled=True,
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:63]: model_compile_enabled is currently doing nothing, I suppose we latter need to regional inductor it.

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py [hunk 3]
```diff
@@ -35,6 +40,31 @@ def graph_trainer_deepseek_v3_debugmodel_ep() -> GraphTrainer.Config:
...
+                # Include-list of FQN substrings. Skips wkv_a/wkv_b (K/V
+                # projections), router.gate, and lm_head.
+                fqns=[
+                    "attention.wq",
+                    "attention.wo",
+                    "feed_forward",
+                    "shared_experts",
+                ],
+            ),
+            MXFP8GroupedExpertsConverter.Config(model_compile_enabled=True),
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:73]: all use of MXFP8GroupedExpertsConverter in titan seem to set recipe_name="mxfp8_rceil". 

not sure what it does, but worth checking

File: torchtitan/experiments/graph_trainer/llama3/config_registry.py [hunk 4]
```diff
@@ -26,6 +28,23 @@ def graph_trainer_llama3_debugmodel() -> GraphTrainer.Config:
...
 
 
+def graph_trainer_llama3_debugmodel_mxfp8() -> GraphTrainer.Config:
+    base = llama3_debugmodel()
+    base.model_spec = llama3_model_registry(
+        "debugmodel",
+        quantization=[
+            MXFP8LinearConverter.Config(
+                model_compile_enabled=True,
+                # Include-list of FQN substrings. Skips wk, wv, lm_head.
```
Comment (SherlockNoMad) [inline torchtitan/experiments/graph_trainer/llama3/config_registry.py:39]: same here, document why they are skipped

**Review by SherlockNoMad — APPROVED** [review]

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py (hunk 2 above)
Comment (aditvenk) [reply torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:63]: it is actually consumed by torchao to print a warning that perf will be bad if compile is disabled

File: torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py (hunk 1 above)
Comment (aditvenk) [reply torchtitan/experiments/graph_trainer/deepseek_v3/config_registry.py:65]: yes, for accuracy reasons. From https://pytorch.org/blog/mxfp8-training-for-moes-1-3x-training-speedup-vs-bf16-for-llama4-scout-on-gb200-cluster-using-torchao-and-torchtitan/

Will add a comment
